docs/src/references/tedge-log-management.md

```diff
@@ -0,0 +1,232 @@
+by creating an empty file at `/etc/tedge/operations/c8y/c8y_LogfileRequest`.
+For other cloud platforms, it does nothing as of now.
+
+### Supported log types
+On startup, the plugin also reports all the log file types that it manages, defined in the `tedge-log-plugin.toml`, to the MQTT topic `te/device/main///cmd/log_upload` with JSON payload,
+as defined in the Configuration section.
+
+## Handling log requests from cloud mapper
+
+After all initialization, this plugin subscribes to `te/device/main///cmd/log_upload/+` topic, listening for 
+log file upload request from any cloud mapper, like this one.
+
+Topic (provided "1234" is operation ID):
+```text
+te/device/main///cmd/log_upload/1234
+```
+
+Payload:
+```json
+{
+  "status": "init",
+  "uploadUrl": "http://127.0.0.1:8000/tedge/file-transfer/main/log_upload/mosquitto",
+  "logType": "mosquitto",
+  "dateFrom": "2013-06-22T17:03:14.000+02:00",
+  "dateTo": "2013-06-22T18:03:14.000+02:00",
+  "searchText": "ERROR",
+  "maximumLines": 1000
+}
+```
+
+The plugin then checks the `tedge-log-plugin.toml` file for the log type in the incoming message (`mosquitto`),
+retrieves the log files using the `target` glob pattern provided in the plugin config file,
+including only the ones modified within the date range(`2013-06-22T17:03:14.000+02:00` to `2013-06-22T18:03:14.000+02:00`),
+with the content filtered by the search text(`ERROR`) and the maximum line count(`1000`).
+
+This filtered content is then uploaded to the given url as `uploadUrl` via HTTP PUT request.
+
+During this process, the plugin need to update the operation status via MQTT for `executing `, `uploading` and `failed` 
+by publishing a message to the topic `te/device/main///cmd/log_upload/<op_id>`.
+
+So, for example, the operation status update message should look like below. The topic is `te/device/main///cmd/log_upload/1234`.
+
+```json
+// Why do we need to repeat other fileds than status and reason?
+{
+  "status": "failed",
+  "reason": "The target log file for 'mosquitto' does not exist."
+  "uploadUrl": "http://127.0.0.1:8000/tedge/file-transfer/main/log_upload/mosquitto",
+  "logType": "mosquitto",
+  "dateFrom": "2013-06-22T17:03:14.000+02:00",
+  "dateTo": "2013-06-22T18:03:14.000+02:00",
+  "searchText": "ERROR",
+  "maximumLines": 1000,
+  "reason": "The type 'typeA' is not configured."
+}
+```
+
+### Successful flow
+
+The status transition: `init` -> `executing` -> `uploading` -> `successful`
+
+```mermaid
+sequenceDiagram
+  participant Cloud
+  participant Mapper
+  participant Plugin
+
+  Cloud->>Mapper: Cloud-specific Log file upload request (Status: init)
+  Mapper->>Plugin: Cloud-agnostic Log file upload request (Status: init)
+  Plugin->>Mapper: Status: executing
+  alt if necessary
+    Mapper->>Cloud: Status: executing
+  end
+  Plugin->>Plugin: Extract log
+  Plugin->>Mapper: File transfer [HTTP]
+  Plugin->>Mapper: Status: uploading
```

Comment:
```suggestion
  Plugin->>Mapper: Status: uploading
  Plugin->>Mapper: File transfer [HTTP]
  Plugin->>Mapper: Status: successful
```
The `uploading` status must be sent before starting the upload, right? No point in sending it after the upload completes. Similarly, the completion of upload must be notified to the mapper either with a `successful` or `failed` status.

Reply:
Since the states will be defined in Didier's state machine definition, I decided to keep status simplest as of now. So,
* Plugin sends only 3 status: executing, successful, and failed.
* The timing of successful is after uploading to http file transfer repository is done. (= that's the end of plugin's one cycle of handling an operation)
* Mapper still needs to upload the log file to cloud, then send a successful message to cloud. However, this "successful" is different from the "successful" from plugin.
I will update accordingly soon.